### ISSUE TITLE
Calculating false positives and other metrics

### ISSUE THREAD

**Issue by FleetingA** [ultralytics/yolov5#4133]
What is the best way to calculate metrics (such as false positives and negatives) on a complete dataset after training is complete? 

I want to calculate how many of the objects of interest in my images have been identified correctly (I have 4700 training images), but am a bit unsure how to derive these metrics. 

Thanks in advance!

**Comment by glenn-jocher** [comment]
@FleetingA you can compute your TP and FP easily from your P and R, i.e. 0.9 recall means that 90% of your objects were correctly identified, and 50% P means that for every TP there was also an FP. 

See https://en.wikipedia.org/wiki/Precision_and_recall

**Comment by glenn-jocher** [comment]
@FleetingA good news 😃! Your original issue may now be fixed ✅ in PR #5727. This PR explicitly computes TP and FP from the existing Labels, P, and R metrics:

```bash
TP = Recall * Labels
FP = TP / Precision - TP
```

These TP and FP per-class vectors are left in val.py for users to access if they want:
https://github.com/ultralytics/yolov5/blob/36d12a500eae4561d09d4955e1b50b12e57bf6c6/val.py#L240

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
